src/Guzzle/MiddleWare/RequestLogger.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+namespace Acquia\ContentHubClient\Guzzle\Middleware;
+
+use GuzzleHttp\Psr7\Request;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Class RequestLogger.
+ */
+class RequestLogger
+{
+
+    /**
+     * @var \Psr\Log\LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * RequestLogger constructor.
+     *
+     * @param \Psr\Log\LoggerInterface $logger
+     */
+    public function __construct(LoggerInterface $logger)
+    {
+        $this->logger = $logger;
```

Comment:
Majority of methods of this class accept Request and/or Response as their argument(s). This means that Request and Response should be considered as a class member instead (especially that your class name is RequestLogger but surprisingly it's not dependent on Request). This also eliminates the decodeResponse method, since you can do it once and for all.